### ISSUE TITLE
Clean up use of toggle buttons

### ISSUE THREAD

**Issue by nolanlawson** [nolanlawson/pinafore#1633]
After long discussions with @marcozehe in [this thread](https://toot.cafe/@nolan/103110863561375513), I think our use of toggle buttons can be improved in this way:

1. The favorite/boost/follow buttons should just be normal buttons that change their label (e.g. "favorite" vs "unfavorite", "follow" vs "unfollow"). The `aria-pressed` attribute is not really necessary if we change the label, and plus this would make the `aria-label` the same as the `title` which is much less confusing from a dev standpoint (less awkward code) and a user standpoint (some screenreaders read both the aria-label and the title).
2. Ditto for the "add content warning" / "add poll" buttons &ndash; we can just change the label to "remove content warning" / "remove poll."
3. Ditto for the reply button, which can be something like "open reply" and "close reply." Additionally, we can use [`aria-expanded`](https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element) to indicate that there is some UI that is being expanded or collapsed.
4. The pin timeline / choose instance UIs should be radio buttons rather than toggle buttons, because logically you can only select one at a time.
5. In the media modal, the toggle buttons should be either radio buttons or nav links, because once again you can only select one at a time.

So in general, it seems there is no reason for us to use toggle buttons anywhere, since I've just listed above all of our toggle buttons. :slightly_smiling_face:

**Comment by MarcoZehe** [comment]
This sounds quite complete to me.

**Comment by nolanlawson** [comment]
With #1647 and #1648 I would consider this issue fixed.

The only thing I _didn't_ do was `aria-expanded`, because I looked through [the docs](https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element) and wasn't quite sure what it would actually add to the "close reply" / "open reply" button.

I also just made the media buttons a list as it was simpler than `aria-current`, which felt odd to me given that the "go left," "go right," "first item," and "second item" buttons were all mixed together. Instead I just put an explicit label saying which one is the current one, and put all the buttons in a list.
